features_desc.md

```diff
@@ -0,0 +1,206 @@
+                    "type": "object",
+                    "properties": {
+                        "location": {
+                            "type": "string",
+                            "description": "The city and state, e.g. San Francisco, CA"
+                        }
+                    },
+                    "required": ["location"]
+                }
+            }
+        }],
+    tool_choice="auto"
+)
+```
+  The output will look like this:
+```python
+  ChatCompletionMessage(
+    content=None,
+    role='assistant',
+    tool_calls=[ToolCall(arguments='{"location": "San Francisco, CA"}', name='get_current_weather'),
+    ToolCall(arguments='{"location": "Tokyo"}',name='get_current_weather'),
+    ToolCall(arguments='{"location": "Paris"}',name='get_current_weather')],
+    function_call=None,
+    tool_call_id=None,
+    name=None
+)
+
+```
+</details>
+
+## Following Up on Missing Function Arguments
+The function can identify when a function call is missing necessary arguments and subsequently ask the user for the required information. This prevents the model from making incorrect assumptions or "hallucinating" data, ensuring that the responses are accurate and based on complete information.
+
+<details>
+  <summary>Example Following Up on Missing Function Arguments (click to expand)</summary>
+  
+```python
+from openai import OpenAI
+
+client = OpenAI(base_url="http://localhost:8000/v1", api_key="functionary")
+
+client.chat.completions.create(
+    model="meetkai/functionary-small-v2.2",
+    messages=[{"role": "user",
+            "content": "What's the weather now?"}
+    ],
+    tools=[{
+            "type": "function",
+            "function": {
+                "name": "get_current_weather",
+                "description": "Get the current weather",
+                "parameters": {
+                    "type": "object",
+                    "properties": {
+                        "location": {
+                            "type": "string",
+                            "description": "The city and state, e.g. San Francisco, CA"
+                        }
+                    },
+                    "required": ["location"]
+                }
+            }
+        }],
+    tool_choice="auto"
+)
+```
+  The output will look like this:
+```python
+  ChatCompletionMessage(
+    content="To provide you with the current weather, I need to know the specific location you're interested in. Could you please provide me with the name of the city and state or country you want to check the weather for?",
+    role='assistant',
+    tool_calls=[],
+    function_call=None,
+    tool_call_id=None,
+    name=None
+)
+
+```
+</details>
+
+## Multi-turn
+The model supports multi-turn conversations. This means the model can have conversations that go back and forth many times, like a real conversation. It maintains the context and continuity of the conversation, allowing for more complex and detailed discussions. 
+
+## Code Interpreter
```

Comment:
We will remove this, because we decided to not include Code interpreter in Version 2.2 as the quality is not good enough, this will come back for version 2.3